src/devices/SoftPwm/SoftwarePwmChannel.cs

```diff
@@ -105,106 +110,89 @@ public SoftwarePwmChannel(int pinNumber, int frequency = 400, double dutyCycle =
                 _shouldDispose = shouldDispose;
             }
 
-            _pin = pinNumber;
-            _controller.OpenPin(_pin, PinMode.Output);
-            _usePrecisionTimer = usePrecisionTimer;
-            _isRunning = false;
+            UpdatePulseWidthParameters();
 
-            _frequency = frequency;
-            _dutyCycle = dutyCycle;
+            _thread = new Thread(Run);
 
-            UpdatePulseWidthParameters();
+            if (usePrecisionTimer)
+            {
+                _stopwatch = new Stopwatch();
+                _stopwatch.Start();
+                _thread.Priority = ThreadPriority.Highest;
+            }
+
+            _controller.OpenPin(_pin, PinMode.Output);
 
-            _runningThread = new Thread(RunSoftPWM);
-            _runningThread.Start();
+            _thread.Start();
         }
 
         private void UpdatePulseWidthParameters()
         {
-            _periodInMilliseconds = 1000.0 / _frequency;
-            _pulseWidthInMilliseconds = _dutyCycle * _periodInMilliseconds;
+            double cycleTicks = TimeSpan.TicksPerSecond / (double)_frequency;
```

Comment:
I'm not quite sure why the old logic used `1000` but the new logic just works based on `TimeSpan Ticks` which are well-defined to be `100ns`